### PR TITLE
Tf 12500/test plan with test refactor

### PR DESCRIPTION
**Pull request by LeoFuna**
### Ticket(s)

_Required unless this is just a change to documentation._

- _[12500](https://airtable.com/appfJZShN8K4tcWHU/pagXdnDYi0tVl4u8R?HjEAY=reck1wP7mlkVgmhNn)_

### Type of change

_Please delete any options that are not relevant_

- [x] Documentation or testing

### Description
We need to enhance our testing doc giving more advices of what do and what to avoid.
We need to have a testing plan to make sure that the initial tests and future tests have some guidance of how to be implemented.
We need to refactor current tests to fit our Test Plan

### Overall PR Checklist:

_Please delete any options that are not relevant_

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if applicable)
- [x] My changes generate no new warnings in the console
- [x] There are no new linting errors/problems in my code
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [x] I have filled out the new 3rd party dependency checklist (if applicable)
- [x] I have filled this PR out fully, and cleaned up any extraneous items so that it is clear and easy to understand

### CODE REVIEW

**Comment by e-papanicolas** [comment]
@LeoFuna I added a commit with some small spelling/grammar changes. Since English is not your first language it felt kind of silly/rude to mark up this amazing PR with those kind of comments, and it was easier to just make the changes quickly myself.

File: cypress/e2e/misdemeanor/fishing/pages/m-offense-fish-2-cont.cy.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  });
+
+  it('when select NO option, open continue to general criteria', () => {
+    cy.visit('/calculator/m-offense-fish-2-cont');
+
+    cy.get('a:visible').contains(/^No$/).click();
+    cy.url().should('include', '/calculator/m-offense-fish-6-cont');
+  });
+
+  it('when select "I\'m not sure" option, open pop-up with some information', () => {
```
Comment (e-papanicolas) [inline cypress/e2e/misdemeanor/fishing/pages/m-offense-fish-2-cont.cy.ts:20]: I see this test on every page, is it necessary to test on every page when this question is always going to be the exact same thing on every page?

**Review by e-papanicolas — COMMENTED** [review]
@LeoFuna I love this! And I appreciate you SO very much for creating this for us before you leave. 

I see one thing that may make this testing a bit difficult, but never impossible -- the information coming from Sanity IS an external dependency in my opinion. I think that the path/slug and page title (which are the same) for our pages, are the only things that are guaranteed to stay static. The content isn't necessarily getting updated constantly, but the point of using Sanity for the content is that anyone, on any team any time is able to make those changes. Completely separately from anything going on in this repository. So what happens if someone makes a change to a word or phrase in the one of the `strong:visible` elements, this will break. Do we need to have an extra step that has people updating content to inform developers to make changes to tests? Or should we just test the paths using only the slug or title, and then also test the other components that should be on the screen? Those things are also theoretically likely to change via Sanity, but configs and things for those are less likely than the other content I think.

File: cypress/README.md [hunk 2]
```diff
@@ -1,21 +1,92 @@
...
+   - Class C felony
+   - Class C felony (with prostitution)
+
+### How to test Calculator
+
+One of our focus points is to try to duplicate code as little as possible, and to do that we can't test every single calculator path because this would cause too much duplicated code.
+
+![Duplication problem](https://github.com/clearviction-devs/clearviction-wa/assets/80538553/82a7ef65-4d7c-4851-bcc8-7df6453b5e48)
+
+We can note that in a hypothetical path, if we try to test all the path from "Entering into Calculator" to "Eligible Page" and from "Entering into Calculator" to "One Ineligible Page", this will cause a duplication of code on `Calculator Home`, `Page1` and `Page2` components, because they have same path. Now imagine this on plenty of paths 😨.
```
[the remark this answers is not in the document]
Comment (LeoFuna) [reply cypress/README.md:76]: I think i got it! Hummm I would say that we can do like this:
1 - We change our `access-calculator` test to be: from `head-initial-1-cont` to `head-3-mis-cont` (which is our first divergence)
2 -  We create 2 new tests: one from `head-3-mis-cont` to `head-special-4-cont` (misdemeanor path) and other from `head-3-mis-cont` to `head-felony-4-cont` (felony path)
With this we make sure that `misdemeanor` and `felony` are accessible.
3 - We create our "happy path" to each option, as i did with `misdemeanor/fishing`.
4 - We creante tests for specific pages to check "non eligible" options and other behaviors.

What do you think?

File: cypress/e2e/misdemeanor/fishing/pages/m-offense-fish-2-cont.cy.ts (hunk 1 above)
Comment (LeoFuna) [reply cypress/e2e/misdemeanor/fishing/pages/m-offense-fish-2-cont.cy.ts:20]: Thats is a good question lol.
As calculator its just a page with changes by `slug`, i think we can just check this in one page and we kinda ensure that it will be working in all other pages.
I think its better to remove those duplications

**Comment by LeoFuna** [comment]
> @LeoFuna I love this! And I appreciate you SO very much for creating this for us before you leave.
> 
> I see one thing that may make this testing a bit difficult, but never impossible -- the information coming from Sanity IS an external dependency in my opinion. I think that the path/slug and page title (which are the same) for our pages, are the only things that are guaranteed to stay static. The content isn't necessarily getting updated constantly, but the point of using Sanity for the content is that anyone, on any team any time is able to make those changes. Completely separately from anything going on in this repository. So what happens if someone makes a change to a word or phrase in the one of the `strong:visible` elements, this will break. Do we need to have an extra step that has people updating content to inform developers to make changes to tests? Or should we just test the paths using only the slug or title, and then also test the other components that should be on the screen? Those things are also theoretically likely to change via Sanity, but configs and things for those are less likely than the other content I think.

@e-papanicolas I really appreciate your review and changes! thx a lot :tada: 

Good point! That is something that brings out the most important questo:  "What in each page is REALLY important?"
Because as you said, if someone change some text in `Sanity`, those `page test` will break.

If the only thing important would be to have a `title` with a `subtitle` with 2 `buttons`, we should not test using a `text`, but using another way, the only situation to test using `text` might be when the content of this text is REALLY important.

With that said, I raise this question: Do you know if the only important data on those pages are: have a title, have subtitle, have 2 buttons etc? Or Do we have pages that have really important texts that we need to ensure that they are wrote in a exactly way?

In short, we need to test only what we think is  important.

File: cypress/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (e-papanicolas) [reply cypress/README.md:76]: Yes this is great.

**Comment by e-papanicolas** [comment]
> > @LeoFuna I love this! And I appreciate you SO very much for creating this for us before you leave.
> > I see one thing that may make this testing a bit difficult, but never impossible -- the information coming from Sanity IS an external dependency in my opinion. I think that the path/slug and page title (which are the same) for our pages, are the only things that are guaranteed to stay static. The content isn't necessarily getting updated constantly, but the point of using Sanity for the content is that anyone, on any team any time is able to make those changes. Completely separately from anything going on in this repository. So what happens if someone makes a change to a word or phrase in the one of the `strong:visible` elements, this will break. Do we need to have an extra step that has people updating content to inform developers to make changes to tests? Or should we just test the paths using only the slug or title, and then also test the other components that should be on the screen? Those things are also theoretically likely to change via Sanity, but configs and things for those are less likely than the other content I think.
> 
> @e-papanicolas I really appreciate your review and changes! thx a lot 🎉
> 
> Good point! That is something that brings out the most important questo: "What in each page is REALLY important?" Because as you said, if someone change some text in `Sanity`, those `page test` will break.
> 
> If the only thing important would be to have a `title` with a `subtitle` with 2 `buttons`, we should not test using a `text`, but using another way, the only situation to test using `text` might be when the content of this text is REALLY important.
> 
> With that said, I raise this question: Do you know if the only important data on those pages are: have a title, have subtitle, have 2 buttons etc? Or Do we have pages that have really important texts that we need to ensure that they are wrote in a exactly way?
> 
> In short, we need to test only what we think is important.

Here are all of the components for the calculator and all of the ways in which we conditionally display different pieces on the actual page. If we also look at the Sanity screens, to see that pages that are marked isQuestion, will include choices, and also will include the I don't know button, and some pages might just have "content". Some pages might not be a question and just have a "continue" single button. So I do think that we need to remove the text from the tests since that is likely to change and so it's maybe not important to test. Do any of these examples help give insight into the important data?

**Comment by e-papanicolas** [comment]
@LeoFuna ok yes so then to be honest I am not 100% sure what the best way to decide what is important or what to test 🫣 I'm inclined to go with whatever way you suggest.

**Comment by LeoFuna** [comment]
> @LeoFuna ok yes so then to be honest I am not 100% sure what the best way to decide what is important or what to test 🫣 I'm inclined to go with whatever way you suggest.

@e-papanicolas  Ok, so lets follow [cypress doc](https://docs.cypress.io/guides/references/best-practices#Selecting-Elements) and try to use `data-cy` and if its not possible, we use `contains('text')`.
And for now, lets just check if things are there, trying to not be too much coupled to Sanity.

I will refactor this PR trying to do this.

**Review by e-papanicolas — APPROVED** [review]
